config-model-api/src/test/java/com/yahoo/config/application/api/TimeWindowTest.java

```diff
@@ -25,71 +28,106 @@
+            assertOutside(tw, i0);
+            assertOutside(tw, i1);
+            assertInside(tw, i2);
+            assertInside(tw, i3);
+            assertOutside(tw, i4);
+            assertOutside(tw, i5);
         }
         {
-            TimeWindow tw = TimeWindow.from("mon", "12,13", "CET");
+            TimeWindow tw = TimeWindow.from("mon", "12,13", "CET", "", "");
             Instant i0 = Instant.parse("2017-09-17T11:15:30.00Z");
             Instant i1 = Instant.parse("2017-09-18T09:15:30.00Z");
             Instant i2 = Instant.parse("2017-09-18T10:15:30.00Z"); // Including offset this matches hour 12
             Instant i3 = Instant.parse("2017-09-18T11:15:30.00Z"); // Including offset this matches hour 13
             Instant i4 = Instant.parse("2017-09-18T12:15:30.00Z");
             Instant i5 = Instant.parse("2017-09-19T11:15:30.00Z");
-            assertFalse("Instant " + i0 + " is not in window", tw.includes(i0));
-            assertFalse("Instant " + i1 + " is not in window", tw.includes(i1));
-            assertTrue("Instant " + i2 + " is in window", tw.includes(i2));
-            assertTrue("Instant " + i3 + " is in window", tw.includes(i3));
-            assertFalse("Instant " + i4 + " is not in window", tw.includes(i4));
-            assertFalse("Instant " + i5 + " is not in window", tw.includes(i5));
+            assertOutside(tw, i0);
+            assertOutside(tw, i1);
+            assertInside(tw, i2);
+            assertInside(tw, i3);
+            assertOutside(tw, i4);
+            assertOutside(tw, i5);
+        }
+        {
+            TimeWindow tw = TimeWindow.from("mon-sun", "0-23", "CET", "2022-01-15", "2022-02-15");
+            Instant i0 = Instant.parse("2022-01-14T12:00:00.00Z"); // Before window
+            Instant i1 = Instant.parse("2022-01-14T23:00:00.00Z"); // Inside window because of time zone offset
+            Instant i2 = Instant.parse("2022-02-05T12:00:00.00Z");
+            Instant i3 = Instant.parse("2022-02-14T23:00:00.00Z");
+            Instant i4 = Instant.parse("2022-02-15T23:00:00.00Z"); // After window because of time zone offset
+            Instant i5 = Instant.parse("2022-02-16T12:00:00.00Z"); // After window
+            assertOutside(tw, i0);
+            assertInside(tw, i1);
+            assertInside(tw, i2);
+            assertInside(tw, i3);
+            assertOutside(tw, i4);
+            assertOutside(tw, i5);
+
+            TimeWindow tw2 = TimeWindow.from("sun", "1", "CET", "2022-01-01", "2022-01-02");
+            Instant i6 = Instant.parse("2022-01-01T00:00:00.00Z"); // Wrong day
+            Instant i7 = Instant.parse("2022-01-02T01:00:00.00Z"); // Wrong hour because of time zone offset
+            Instant i8 = Instant.parse("2022-01-02T00:00:00.00Z");
+            assertOutside(tw2, i6);
+            assertOutside(tw2, i7);
+            assertInside(tw2, i8);
+
+            TimeWindow tw3 = TimeWindow.from("", "", "CET", "2022-01-02", "");
```

Comment:
Its not possible to define any range shorter than 7 days without also specifying `days`

Reply:
Thanks! Fixed in https://github.com/vespa-engine/vespa/pull/20699